### PR TITLE
Fix issue with autoscaler

### PR DESCRIPTION
**Pull request by davidrissato**
# Description

Autoscaler is being terminated with an exception during `count_running_alignment_jobs` execution.

# Notes

This error is not easy to reproduce because it depends on a combination of current time and specific services responses.

# Tests

* Created a unit test that can trigger the issue.

### CODE REVIEW

File: app/jobs/autoscaling.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ def discard_if_safe(self):
         def count_running_alignment_jobs():
             ''' returns a map of draining instance IDs to number of jobs still running on the instance '''
             instance_ids = self.draining_instances
-            count_dict = {id: 0 for id in instance_ids}
+            count_dict = defaultdict(int, {id: 0 for id in instance_ids})
```
Comment (cdebourcy) [inline app/jobs/autoscaling.py]: Thanks for making this change! To follow up on our offline discussion: the function count_running_alignment_jobs will count across all instances. But only the counts from draining instances will actually be used to make discard decisions due to line https://github.com/chanzuckerberg/idseq-web/blob/acf90afc37423de2571ca8d97a7584e356cb54cc/app/jobs/autoscaling.py#L413. So the counts for non-draining instances are unnecessary. If you wanted, you could also fix the key error by wrapping line https://github.com/chanzuckerberg/idseq-web/blob/acf90afc37423de2571ca8d97a7584e356cb54cc/app/jobs/autoscaling.py#L402 into a "if key present" type condition. But the defaultdict solution seems less invasive

Comment (davidrissato) [reply app/jobs/autoscaling.py]: Actually I did exactly that in my first attempt, but I ended up amending my commit with the `defaultdict` approach to be more aligned to the method name.

Comment (davidrissato) [reply app/jobs/autoscaling.py]: There are some conflicts in this branch. Let me fix these issues, and maybe I'll add the filter "if key present" as well.

Comment (cdebourcy) [reply app/jobs/autoscaling.py]: Yes I agree I actually think your current fix with the defaultdict is better to satisfy the method name, no need to change!

Comment (davidrissato) [reply app/jobs/autoscaling.py]: This method actually does a little bit more than just counting. It removes expired tags from draining instances as well, and the method comment doesn't necessarily match its functionality anymore.
I wanted to refactor the entire method to decouple this tag deletion logic, but considering this code is somewhat stable, I'm postponing a refactoring on it to a moment that we need to change other things on that logic.

**Pull request merged**